### ISSUE TITLE
Incorrect columns in the result of read_excel

### ISSUE THREAD

**Issue by dchigarev** [modin-project/modin#1733]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Modin version** (`modin.__version__`): 0.7.3+200.g8e7a5682
- **Python version**: 3.7.5
- **Code we can use to reproduce**:
```python
if __name__ == "__main__":
    import modin.pandas as pd
    import pandas
    from modin.pandas.test.test_io import setup_excel_file, TEST_EXCEL_FILENAME

    setup_excel_file(5, force=True)

    pd_res = pandas.read_excel(TEST_EXCEL_FILENAME)
    md_res = pd.read_excel(TEST_EXCEL_FILENAME)

    print("pd_result:\n", pd_res, sep="")
    print("\nmd_result:\n", md_res, sep="")

    print("\npd_columns:", pd_res.columns)
    print("md_columns:", md_res.columns)

```

<details><summary>Output:</summary>

```
pd_result:
   Unnamed: 0  col1  col2
0           0     0     0
1           1     1     1
2           2     2     2
3           3     3     3
4           4     4     4

md_result:
   Unnamed: 0  col1  col2
0           0     0     0
1           1     1     1
2           2     2     2
3           3     3     3
4           4     4     4

pd_columns: Index(['Unnamed: 0', 'col1', 'col2'], dtype='object')
md_columns: Index(['', 'col1', 'col2'], dtype='object')
```

</details>

### Describe the problem
Columns into partitions seems to be correct, but columns of dataframe itself isn't, that's also the reason why test don't fails on that test case (`to_pandas` that used in `df_equals` considers information only from partitions, fixing it in #1727)

@devin-petersohn have you familiar with `read_excel` implementation, is this has a quick fix?

**Comment by devin-petersohn** [comment]
@dchigarev, it can be a quick fix. We are using pandas for determining columns, so there inconsistencies with how pandas does this as well. I will investigate.

**Comment by vnlitvinov** [comment]
I can no longer reproduce the issue.
I think the original issue had the same root cause as #2404 which I had earlier fixed by #2526, so #1733 should be fixed as well.

I'm now just re-enabling the test disabled due to this issue in PR #2838.
